### PR TITLE
feature: Make KVM work with Garden Linux ARM

### LINKED ISSUE #1040: ARM64: PyTest -> Integration Tests -> `KVM`
Meta Issue: #1028 

## PyTest
This issue covers the validation of `integration tests` of PyTest for the platform provider `KVM`  running on `ARM64` / `AARCH64` architectures.

## Tests
Within this issue the following components will be validated and tested:
- PyTest
  - Integration Tests
    - `KVM`

### PR DESCRIPTION
**Pull request by nanory**
**How to categorize this PR?**
/kind bug
/area os
/os garden-linux

**What this PR does / why we need it**:
This PR fixes the KVM platform and some tests which failed on the `kvm` Garden Linux image using the `kvm` pytests.
In addition, the `kvm` platform has been modified, so that it uses the `bin/start-vm` script to spawn a KVM machine instead of using its own Qemu command. However, the `start-vm` script has been adjusted as well in order to work with the `kvm` platform by introducing the new CLI parameter `--destport`.

**Which issue(s) this PR fixes**:
Fixes #1040 

**Special notes for your reviewer**:
The `tiger` tests has been disabled by this PR since it causes issues with other tests that are executed afterwards. In order to fix `tiger`, the following issue has been created: #1102

### CODE REVIEW

File: bin/start-vm [hunk 1]
```diff
@@ -449,7 +453,7 @@ printf "  starting VM(UUID:%s) with MAC:%s in %s\n" $uuid $macfull $targetDir
 [ $pxe ]	&& printf "  pxeboot: %s.ipxe\n" $mac
 [ $pxe ]	&& printf "  pxeboot: files served from %s\n" "$pxe"
 [ $bridge ]	&& printf "  interface: %s  bridged\n" $bridge
-[ $bridge ]	|| printf "  sshport: %s  unbridged\n" $port
+[ $bridge ]	|| printf "  sshport: %s (%s) unbridged\n" $port $port_dest
```
Comment (gyptazy) [inline bin/start-vm:456]: we should shortly describe the differences within the output, while only `support: 2223 (22) unbridged` could be a little bit confusing. Maybe:

`printf " sshport: VM tcp/%s mapped to host tcp/%s unbridged\n" $port_dest $port`
or
`printf " sshport: local %s  -> vm %s (unbridged)\n" $port $port_dest`

File: tests/integration/kvm.py [hunk 2]
```diff
@@ -176,112 +178,102 @@ def _adjust_kvm(self):
...
+                image_name=image_name,
+                authorized_keys_file=authorized_keys_file,
+                sshd_systemd_file=sshd_systemd_file,
+                sshd_config_file=sshd_config_file
+            )
+        )
+        cmds.append(copy_cmd)
+
+        # Modify the snapshot via guestfish
+        guestfish_cmd = (
```
Comment (gyptazy) [inline tests/integration/kvm.py:209]: Probably, this is a good time to get rid of this old format of `"{foo}".format(foo=bar)` and switch to `f"{bar}"`

**Review by gyptazy — CHANGES_REQUESTED** [review]

File: build.sh [hunk 3]
```diff
@@ -184,7 +184,7 @@ else
 		fullfeatures="$(cat $outputDir/fullfeature.info)"
 		configDir=$(${thisDir}/bin/garden-integration-test-config kvm ${prefix} ${fullfeatures} ${outputDir} ${arch})
 		echo "Running pytests in KVM"
-		${gardenlinux_build_cre} run --name $containerName --rm -v /boot/:/boot \
+		${gardenlinux_build_cre} run --name $containerName --privileged --rm -v /boot/:/boot \
```
Comment (nkraetzschmar) [inline build.sh:187]: why do we need `--privileged` now?

Comment (nkraetzschmar) [reply build.sh:187]: shouldn't `--device=/dev/kvm` be sufficient?

**Comment by chrinorse** [comment]
Please do not remove the tiger test, the test works as expected in the chroot test environment and should only be excluded from other test environments like kvm.

**Comment by nanory** [comment]
> Please do not remove the tiger test, the test works as expected in the chroot test environment and should only be excluded from other test environments like kvm.
I re-added the tiger test since it is only executed on `chroot` from now on, so it shouldn't cause any further problems.

**Review by gyptazy — APPROVED** [review]
technically lgtm

While we could change all `"{foo}".format(foo=bar)` to `f"{bar}"` here for `kvm.py`, I'm also fine with creating a new issue to proceed on this because this could be out of scope here. What do you think?

**Review by chrinorse — APPROVED** [review]
lgtm

**Pull request merged**